src_c/font.c

```diff
@@ -878,13 +950,19 @@ font_metrics(PyObject *self, PyObject *textobj)
     for (i = 1 /* skip BOM */; i < length; i++) {
         ch = buffer[i];
         surrogate = Py_UNICODE_IS_SURROGATE(ch);
-        /* TODO:
-         * TTF_GlyphMetrics() seems to return a value for any character,
-         * using the default invalid character, if the char is not found.
-         */
+/* TODO:
+ * TTF_GlyphMetrics() seems to return a value for any character,
+ * using the default invalid character, if the char is not found.
+ */
+#if SDL_TTF_VERSION_ATLEAST(3, 0, 0)
+        if (!surrogate && /* conditional and */
+            !TTF_GetGlyphMetrics(font, (Uint16)ch, &minx, &maxx, &miny, &maxy,
```

Comment:
```suggestion
            TTF_GetGlyphMetrics(font, ch, &minx, &maxx, &miny, &maxy,
```
Is the SDL error set if the call fails. If yes, should it be cleared? The cast to (Uint16) seem to be pointless. Also in SDL3, Uint32 is used, in SDL2 there's ``TTF_GlyphMetrics32``. Is there any reason to use (Uint16) instead of (Uint32)?

Reply:
Well, `TTF_GlyphMetrics32` was added in SDL 2.0.18, while our minimum supported version at the moment is SDL 2.0.15. The fix for this is a bit more involved and probably out of scope of this PR anyways, but I will look into it in another PR